src/cli/self_update.rs

```diff
@@ -0,0 +1,281 @@
+    let binary_tempdir = &tempfile::tempdir().into_diagnostic()?;
+
+    // Uncompress the archive
+    if archive_name.ends_with(".tar.gz") {
+        let mut archive = Archive::new(GzDecoder::new(archived_tempfile.as_file()));
+        archive.unpack(binary_tempdir).into_diagnostic()?;
+    } else if archive_name.ends_with(".zip") {
+        let mut archive = zip::ZipArchive::new(archived_tempfile.as_file()).into_diagnostic()?;
+        archive.extract(binary_tempdir).into_diagnostic()?;
+    } else {
+        println!(
+            "{}Unsupported archive format: {}",
+            console::style(console::Emoji("✘ ", "")).red(),
+            archive_name
+        );
+        std::process::exit(1);
+    }
+
+    println!(
+        "{}Pixi archive uncompressed.",
+        console::style(console::Emoji("✔ ", "")).green(),
+    );
+
+    // Get the new binary path used for self-replacement
+    let new_binary_path = binary_tempdir.path().join(pixi_binary_name());
+
+    // Replace the current binary with the new binary
+    self_replace::self_replace(new_binary_path).into_diagnostic()?;
+
+    println!(
+        "{}Pixi has been updated to version {}.",
+        console::style(console::Emoji("✔ ", "")).green(),
+        target_version
+    );
+
+    Ok(())
+}
+
+async fn retrieve_target_version(version: &Option<String>) -> GithubRelease {
+    // Fetch the target version from github.
+    // The target version is:
+    // - the latest version if no version is specified
+    // - the specified version if a version is specified
+    let url = if let Some(version) = version {
+        format!(
+            "https://api.github.com/repos/prefix-dev/pixi/releases/tags/v{}",
+            version
+        )
+    } else {
+        "https://api.github.com/repos/prefix-dev/pixi/releases/latest".to_string()
+    };
+
+    let client = Client::new();
+
+    let res = client
+        .get(url)
+        .header("User-Agent", user_agent())
+        .send()
+        .await
+        .expect("Failed to fetch latest version from github");
+
+    let body = res
+        .text()
+        .await
+        .expect("Failed to fetch latest version from github");
+
+    // compare target version with current version
+    let target_version_json = serde_json::from_str::<GithubRelease>(&body);
+
+    match target_version_json {
+        Ok(target_version_json) => target_version_json,
+        Err(err) => {
+            match version {
+                Some(version) => println!(
+                    "{}The version you specified is not available: {}",
+                    console::style(console::Emoji("✘ ", "")).red(),
+                    version
+                ),
+                None => println!(
+                    "{}Failed to fetch latest version from github: {}",
+                    console::style(console::Emoji("✘ ", "")).red(),
+                    err
+                ),
+            }
+            std::process::exit(1);
+        }
+    }
+}
```

Comment:
You can use https://doc.rust-lang.org/std/env/consts/constant.EXE_EXTENSION.html

Reply:
good catch - will use `std::env::consts::EXE_SUFFIX`